src/pages/SignUp/ConfirmForm.tsx

```diff
@@ -0,0 +1,117 @@
+
+type Props = {
+  codeRecipientEmail: CodeRecipientEmail;
+  setSignupState: StateSetter;
+  userType: UserType;
+};
+
+export default function ConfirmForm(props: Props) {
+  const { handleError } = useErrorContext();
+  const [isRequestingNewCode, setIsRequestingNewCode] = useState(false);
+  const methods = useForm({
+    resolver: yupResolver(object({ code: requiredString })),
+  });
+  const {
+    handleSubmit,
+    formState: { isSubmitting },
+  } = methods;
+
+  type FV = typeof methods extends UseFormReturn<infer U> ? U : never;
+
+  const { counter, resetCounter } = useCounter(MAX_TIME);
+
+  const submit = async (fv: FV) => {
+    try {
+      const result = await confirmSignUp({
+        username: props.codeRecipientEmail.raw,
+        confirmationCode: fv.code,
+      });
+
+      if (result.nextStep.signUpStep !== "DONE" || !result.isSignUpComplete) {
+        return handleError("Code confirmation failed", undefined, {
+          log: false,
+        });
+      }
+
+      props.setSignupState({ type: "success", userType: props.userType });
+    } catch (err) {
+      const message =
+        err instanceof AuthError ? err.message : GENERIC_ERROR_MESSAGE;
+      handleError(err, message, { log: !(err instanceof AuthError) });
+    }
+  };
+
+  const resendOTP = async () => {
+    try {
+      setIsRequestingNewCode(true);
+
+      //no need to inspect result
+      await resendSignUpCode({
+        username: props.codeRecipientEmail.raw,
+      });
+      resetCounter();
+
+      alert("New code has been sent to your email.");
+    } catch (err) {
+      const message =
+        err instanceof AuthError ? err.message : GENERIC_ERROR_MESSAGE;
+      handleError(err, message, { log: !(err instanceof AuthError) });
+    } finally {
+      setIsRequestingNewCode(false);
+    }
+  };
+
+  return (
+    <Form
+      className="grid w-full max-w-md px-7 py-8 bg-white border border-gray-l4 rounded-2xl"
+      disabled={isSubmitting || isRequestingNewCode}
+      methods={methods}
+      onSubmit={handleSubmit(submit)}
+    >
+      <h3 className="text-center text-2xl font-bold text-navy-d4">
+        Verify your account
+      </h3>
+      <p className="text-center font-normal mt-2">
+        You’re almost there! 6-digit security code has been sent to{" "}
+        <span className="font-medium">{props.codeRecipientEmail.obscured}</span>
+      </p>
+      <Field<FV>
+        name="code"
+        placeholder="Enter 6-digit code"
+        classes={{ container: "my-6" }}
+      />
+      <button
+        type="submit"
+        className="flex-center bg-blue-d1 disabled:bg-gray text-white enabled:hover:bg-blue enabled:active:bg-blue-d2 h-[52px] rounded-full normal-case text-lg font-bold w-full"
+      >
+        Verify account
+      </button>
+
+      <span className="grid grid-cols-[4fr_3fr] items-center justify-items-start gap-1 text-sm font-medium mt-5">
+        <span className="justify-self-end">Resend code in</span>
+        <button
```

Comment:
Resend inhibitor works - just not sure the numbers be the clickable 
<img width="601" alt="image" src="https://github.com/AngelProtocolFinance/angelprotocol-web-app/assets/89639563/2d7c2ec7-18b5-4d9d-b868-4bdc19bedd88">

would leave to internal team if this is telling enough

Reply:
@SovereignAndrey need your input ser.

Reply:
@misicnenad Will try to find other example and see how they do it and have an example to show. 

My first thought is that the numbers shouldn't be clickable, but that the "resend code" text before it would be clickable (disabled until countdown expires).

~~ OR ~~

It's all text and a "resend code" link/button appears below the text after the timer reaches zero and disappears after clicked when timer restarts.

Reply:
@SovereignAndrey something like this?
![image](https://github.com/AngelProtocolFinance/angelprotocol-web-app/assets/19427053/a7611ee1-6108-4d34-af3e-f369f5418b9b)

Reply:
@misicnenad This is exactly what I had in mind. :100: I'd say place the re-send countdown & link _above_ the verity button.

Reply:
![image](https://github.com/AngelProtocolFinance/angelprotocol-web-app/assets/19427053/d9ac21f2-4515-40f6-9146-e6c56475251c)